Versions of GNAT older than 5 permitted the mistake of not pairing
"limited" from forward declaration.  This fixes numerous instances of that
mistake in qtada.

--- examples/designer/custom_widget_plugin/analog_clock_plugin.ads.orig	2011-01-24 14:14:50 UTC
+++ examples/designer/custom_widget_plugin/analog_clock_plugin.ads
@@ -37,7 +37,7 @@ package Analog_Clock_Plugin is
 private
 
    type Analog_Clock_Plugin is
-     new Qt4.Objects.Directors.Q_Object_Director
+     limited new Qt4.Objects.Directors.Q_Object_Director
        and Qt4.Designer_Custom_Widget_Interfaces.Q_Designer_Custom_Widget_Interface
          with record
       Q_Designer_Custom_Widget_Interface_View :
--- examples/graphics_view/elastic_nodes/graph_widgets.ads.orig	2009-04-03 13:20:32 UTC
+++ examples/graphics_view/elastic_nodes/graph_widgets.ads
@@ -43,7 +43,7 @@ package Graph_Widgets is
 private
 
    type Graph_Widget is
-     new Qt4.Graphics_Views.Directors.Q_Graphics_View_Director with
+     limited new Qt4.Graphics_Views.Directors.Q_Graphics_View_Director with
    record
       Timer_Id    : Qt4.Q_Integer := 0;
 
--- examples/item_views/chart/main_windows.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/item_views/chart/main_windows.ads
@@ -49,7 +49,7 @@ package Main_Windows is
 private
 
    type Main_Window is
-     new Qt4.Main_Windows.Directors.Q_Main_Window_Director with
+     limited new Qt4.Main_Windows.Directors.Q_Main_Window_Director with
    record
       Model           : Qt4.Abstract_Item_Models.Q_Abstract_Item_Model_Access;
       Pie_Chart       : Qt4.Abstract_Item_Views.Q_Abstract_Item_View_Access;
--- examples/item_views/chart/pie_views.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/item_views/chart/pie_views.ads
@@ -49,7 +49,7 @@ package Pie_Views is
 
 private
 
-   type Pie_View is
+   type Pie_View is limited
      new Qt4.Abstract_Item_Views.Directors.Q_Abstract_Item_View_Director with
    record
       Margin      : Qt4.Q_Integer;
--- examples/main_windows/dock_widgets/main_windows.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/main_windows/dock_widgets/main_windows.ads
@@ -39,7 +39,7 @@ package Main_Windows is
 
 private
 
-   type Main_Window is
+   type Main_Window is limited
      new Qt4.Main_Windows.Directors.Q_Main_Window_Director with
    record
       Text_Edit       : Qt4.Text_Edits.Q_Text_Edit_Access;
--- examples/qt_ada/tasking/emitters.ads.orig	2009-12-21 06:36:28 UTC
+++ examples/qt_ada/tasking/emitters.ads
@@ -39,7 +39,7 @@ package Emitters is
 
 private
 
-   type Emitter is
+   type Emitter is limited
      new Qt4.Objects.Directors.Q_Object_Director with null record;
 
    procedure Emit_Progress_Value_Changed
--- examples/qt_ada/tasking/main_windows.ads.orig	2009-12-21 06:36:28 UTC
+++ examples/qt_ada/tasking/main_windows.ads
@@ -37,7 +37,7 @@ package Main_Windows is
 
 private
 
-   type Main_Window is
+   type Main_Window is limited
      new Qt4.Main_Windows.Directors.Q_Main_Window_Director with record
       Progress_Bar : Qt4.Progress_Bars.Q_Progress_Bar_Access;
       Start_Button : Qt4.Push_Buttons.Q_Push_Button_Access;
--- examples/sql/cached_table/table_editors.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/sql/cached_table/table_editors.ads
@@ -43,7 +43,7 @@ package Table_Editors is
 
 private
 
-   type Table_Editor is
+   type Table_Editor is limited
      new Qt4.Dialogs.Directors.Q_Dialog_Director with record
       Submit_Button : Qt4.Push_Buttons.Q_Push_Button_Access;
       Revert_Button : Qt4.Push_Buttons.Q_Push_Button_Access;
--- examples/tools/echo_window/echo_plugin/echo_plugin.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tools/echo_window/echo_plugin/echo_plugin.ads
@@ -45,7 +45,7 @@ package Echo_Plugin is
 
 private
 
-   type Echo_Plugin_Impl is new Qt4.Objects.Directors.Q_Object_Director
+   type Echo_Plugin_Impl is limited new Qt4.Objects.Directors.Q_Object_Director
      and Echo_Interfaces.Echo_Interface with null record;
    pragma Q_Export_Plugin (Echo_Plugin_Impl, Constructors.Create);
 
--- examples/tools/echo_window/echo_windows.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tools/echo_window/echo_windows.ads
@@ -38,7 +38,7 @@ package Echo_Windows is
 
 private
 
-   type Echo_Window is
+   type Echo_Window is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with record
       Echo      : Echo_Interfaces.Echo_Interface_Access;
       Label     : Qt4.Labels.Q_Label_Access;
--- examples/tutorial/t10/cannon_fields.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t10/cannon_fields.ads
@@ -52,7 +52,8 @@ package Cannon_Fields is
 
 private
 
-   type Cannon_Field is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type Cannon_Field is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Current_Angle : Qt4.Q_Integer;
       Current_Force : Qt4.Q_Integer;
    end record;
--- examples/tutorial/t10/lcd_ranges.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t10/lcd_ranges.ads
@@ -46,7 +46,8 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type LCD_Range is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Slider : Qt4.Sliders.Q_Slider_Access;
    end record;
 
--- examples/tutorial/t10/my_widgets.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t10/my_widgets.ads
@@ -33,7 +33,7 @@ package My_Widgets is
 
 private
 
-   type My_Widget is
+   type My_Widget is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with null record;
 
 end My_Widgets;
--- examples/tutorial/t11/cannon_fields.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t11/cannon_fields.ads
@@ -57,7 +57,8 @@ package Cannon_Fields is
 
 private
 
-   type Cannon_Field is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type Cannon_Field is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Current_Angle    : Qt4.Q_Integer;
       Current_Force    : Qt4.Q_Integer;
 
--- examples/tutorial/t11/lcd_ranges.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t11/lcd_ranges.ads
@@ -46,7 +46,7 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is
+   type LCD_Range is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with record
       Slider : Qt4.Sliders.Q_Slider_Access;
    end record;
--- examples/tutorial/t11/my_widgets.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t11/my_widgets.ads
@@ -33,7 +33,7 @@ package My_Widgets is
 
 private
 
-   type My_Widget is
+   type My_Widget is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with null record;
 
 end My_Widgets;
--- examples/tutorial/t12/cannon_fields.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t12/cannon_fields.ads
@@ -63,7 +63,8 @@ package Cannon_Fields is
 
 private
 
-   type Cannon_Field is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type Cannon_Field is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Current_Angle    : Qt4.Q_Integer;
       Current_Force    : Qt4.Q_Integer;
       Timer_Count      : Qt4.Q_Integer;
--- examples/tutorial/t12/lcd_ranges.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t12/lcd_ranges.ads
@@ -46,7 +46,8 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type LCD_Range is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Slider : Qt4.Sliders.Q_Slider_Access;
    end record;
 
--- examples/tutorial/t12/my_widgets.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t12/my_widgets.ads
@@ -33,7 +33,7 @@ package My_Widgets is
 
 private
 
-   type My_Widget is
+   type My_Widget is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with null record;
 
 end My_Widgets;
--- examples/tutorial/t13/cannon_fields.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t13/cannon_fields.ads
@@ -73,7 +73,8 @@ package Cannon_Fields is
 
 private
 
-   type Cannon_Field is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type Cannon_Field is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Current_Angle    : Qt4.Q_Integer;
       Current_Force    : Qt4.Q_Integer;
       Timer_Count      : Qt4.Q_Integer;
--- examples/tutorial/t13/game_boards.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t13/game_boards.ads
@@ -50,7 +50,8 @@ package Game_Boards is
 
 private
 
-   type Game_Board is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type Game_Board is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Hits         : Qt4.LCD_Numbers.Q_LCD_Number_Access;
       Shots_Left   : Qt4.LCD_Numbers.Q_LCD_Number_Access;
       Cannon_Field : Cannon_Fields.Cannon_Field_Access;
--- examples/tutorial/t13/lcd_ranges.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t13/lcd_ranges.ads
@@ -57,7 +57,8 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type LCD_Range is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Slider : Qt4.Sliders.Q_Slider_Access;
       Label  : Qt4.Labels.Q_Label_Access;
    end record;
--- examples/tutorial/t14/cannon_fields.ads.orig	2009-04-09 19:05:06 UTC
+++ examples/tutorial/t14/cannon_fields.ads
@@ -84,7 +84,8 @@ package Cannon_Fields is
 
 private
 
-   type Cannon_Field is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type Cannon_Field is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Current_Angle    : Qt4.Q_Integer;
       Current_Force    : Qt4.Q_Integer;
       Timer_Count      : Qt4.Q_Integer;
--- examples/tutorial/t14/game_boards.ads.orig	2009-04-09 19:05:06 UTC
+++ examples/tutorial/t14/game_boards.ads
@@ -53,7 +53,8 @@ package Game_Boards is
 
 private
 
-   type Game_Board is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type Game_Board is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Hits         : Qt4.LCD_Numbers.Q_LCD_Number_Access;
       Shots_Left   : Qt4.LCD_Numbers.Q_LCD_Number_Access;
       Cannon_Field : Cannon_Fields.Cannon_Field_Access;
--- examples/tutorial/t14/lcd_ranges.ads.orig	2009-04-09 19:05:06 UTC
+++ examples/tutorial/t14/lcd_ranges.ads
@@ -62,7 +62,8 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type LCD_Range is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Slider : Qt4.Sliders.Q_Slider_Access;
       Label  : Qt4.Labels.Q_Label_Access;
    end record;
--- examples/tutorial/t4/my_widgets.ads.orig	2009-04-04 16:14:18 UTC
+++ examples/tutorial/t4/my_widgets.ads
@@ -38,7 +38,8 @@ package My_Widgets is
 
 private
 
-   type My_Widget is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type My_Widget is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Quit : Qt4.Push_Buttons.Q_Push_Button_Access;
    end record;
 
--- examples/tutorial/t5/my_widgets.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t5/my_widgets.ads
@@ -36,7 +36,8 @@ package My_Widgets is
 
 private
 
-   type My_Widget is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type My_Widget is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Quit   : Qt4.Push_Buttons.Q_Push_Button_Access;
       LCD    : Qt4.LCD_Numbers.Q_LCD_Number_Access;
       Slider : Qt4.Sliders.Q_Slider_Access;
--- examples/tutorial/t6/lcd_ranges.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t6/lcd_ranges.ads
@@ -33,7 +33,7 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is
+   type LCD_Range is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with null record;
 
 end LCD_Ranges;
--- examples/tutorial/t6/my_widgets.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t6/my_widgets.ads
@@ -33,7 +33,7 @@ package My_Widgets is
 
 private
 
-   type My_Widget is
+   type My_Widget is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with null record;
 
 end My_Widgets;
--- examples/tutorial/t7/lcd_ranges.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t7/lcd_ranges.ads
@@ -39,7 +39,8 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type LCD_Range is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Slider : Qt4.Sliders.Q_Slider_Access;
    end record;
 
--- examples/tutorial/t7/my_widgets.ads.orig	2009-04-04 16:40:39 UTC
+++ examples/tutorial/t7/my_widgets.ads
@@ -37,7 +37,7 @@ package My_Widgets is
 
 private
 
-   type My_Widget is
+   type My_Widget is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with null record;
 
 end My_Widgets;
--- examples/tutorial/t8/cannon_fields.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t8/cannon_fields.ads
@@ -48,7 +48,8 @@ package Cannon_Fields is
 
 private
 
-   type Cannon_Field is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type Cannon_Field is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Current_Angle : Qt4.Q_Integer;
    end record;
 
--- examples/tutorial/t8/lcd_ranges.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t8/lcd_ranges.ads
@@ -46,7 +46,8 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type LCD_Range is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Slider : Qt4.Sliders.Q_Slider_Access;
    end record;
 
--- examples/tutorial/t8/my_widgets.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t8/my_widgets.ads
@@ -33,7 +33,7 @@ package My_Widgets is
 
 private
 
-   type My_Widget is
+   type My_Widget is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with null record;
 
 end My_Widgets;
--- examples/tutorial/t9/cannon_fields.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t9/cannon_fields.ads
@@ -43,7 +43,7 @@ package Cannon_Fields is
 
 private
 
-   type Cannon_Field is
+   type Cannon_Field is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with record
       Current_Angle : Qt4.Q_Integer;
    end record;
--- examples/tutorial/t9/lcd_ranges.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t9/lcd_ranges.ads
@@ -46,7 +46,8 @@ package LCD_Ranges is
 
 private
 
-   type LCD_Range is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type LCD_Range is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Slider : Qt4.Sliders.Q_Slider_Access;
    end record;
 
--- examples/tutorial/t9/my_widgets.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/tutorial/t9/my_widgets.ads
@@ -33,7 +33,7 @@ package My_Widgets is
 
 private
 
-   type My_Widget is
+   type My_Widget is limited
      new Qt4.Widgets.Directors.Q_Widget_Director with null record;
 
 end My_Widgets;
--- examples/widgets/line_edits/my_widgets.ads.orig	2009-04-18 09:27:41 UTC
+++ examples/widgets/line_edits/my_widgets.ads
@@ -34,7 +34,8 @@ package My_Widgets is
 
 private
 
-   type My_Widget is new Qt4.Widgets.Directors.Q_Widget_Director with record
+   type My_Widget is limited
+     new Qt4.Widgets.Directors.Q_Widget_Director with record
       Access_Line_Edit     : Qt4.Line_Edits.Q_Line_Edit_Access;
       Alignment_Line_Edit  : Qt4.Line_Edits.Q_Line_Edit_Access;
       Echo_Line_Edit       : Qt4.Line_Edits.Q_Line_Edit_Access;
